druid-shell/src/platform/windows/dialog.rs

```diff
@@ -119,7 +121,27 @@ pub(crate) unsafe fn get_file_dialog_path(
 
     if let Some(spec) = &raw_spec {
         as_result(file_dialog.SetFileTypes(spec.len() as u32, spec.as_ptr()))?;
-        as_result(file_dialog.SetFileTypeIndex(1))?;
+
+        let index = match &options.default_type {
+            Some(default_type) => options
+                .allowed_types
+                .as_ref()
+                .unwrap()
```

Comment:
this will panic if `allowed_types` is None, which is possible I think?

Reply:
`allowed_types` is never `None` in this section. However it can be `Some(vec![])` for which I added an additional check. The previous code with the hardcoded `SetFileTypeIndex(1)` call also failed to account for the empty vector case.

This is an interesting issue in a more broad sense. Using `unwrap` in the current situation is fine and doing all the checks would result in a bunch of bloat that makes the code harder to read and maintain. However someone else might modify this function some years down the line and not fully realize the assumptions made here about `None` not being a possibility.

For now I added some comments to clarify the assumptions and care that should be taken with future edits.